File: auth-application/src/main/java/com/example/auth/application/port/in/RevokeSessionUseCase.java
package com.example.auth.application.port.in;

import com.example.auth.domain.common.TenantId;
import com.example.auth.domain.common.UserId;
import java.util.UUID;

public interface RevokeSessionUseCase {

    void revoke(Command cmd);

    record Command(TenantId tenantId, UserId userId, UUID sessionId, String ipAddress) {
    }
}

File: auth-application/src/main/kotlin/com/example/auth/application/port/in/RevokeSessionUseCase.kt
package com.example.auth.application.port.`in`

import com.example.auth.domain.common.TenantId
import com.example.auth.domain.common.UserId
import java.util.UUID

interface RevokeSessionUseCase {

    fun revoke(cmd: Command)

    @JvmRecord
    data class Command(
        val tenantId: TenantId,
        val userId: UserId,
        val sessionId: UUID,
        val ipAddress: String?,
    )
}
